janus_server/src/aggregator.rs

```diff
@@ -799,6 +804,20 @@ impl VdafOps {
                             state: share_data.agg_state.clone(),
                         })
                         .await?;
+
+                        if let ReportAggregationState::<A>::Finished(ref output_share) =
+                            share_data.agg_state
+                        {
+                            Self::put_or_update_batch_unit_aggregation::<A, _>(
+                                tx,
+                                task_id,
+                                min_batch_duration,
+                                share_data.report_share.nonce,
+                                &aggregation_job.as_ref().aggregation_param,
+                                output_share,
+                            )
+                            .await?;
```

Comment:
This case is a 0-round VDAF (or a DAF, I suppose), where an output share can be recovered without any peer prepare message. We don't currently exercise this because we don't have any 0-round VDAFs. I think we should consider adding a variant `task::Vdaf::FakeZeroRound` and using `aggregator::fake::Vdaf` to implement it, but I opted not to do that in this PR which is already a bit large, especially since the lack of test coverage for 0-round VDAFs was a problem before this change.